clang/include/clang/Driver/Options.td

```diff
@@ -7348,9 +7348,14 @@ defm sycl_allow_device_image_dependencies: BoolOptionWithoutMarshalling<"f", "sy
   PosFlag<SetTrue, [], [ClangOption], "Allow dependencies between device code images">,
   NegFlag<SetFalse, [], [ClangOption], "Do not allow dependencies between device code images (default)">>;
 
-def fsycl_dump_device_code_EQ : Joined<["-"], "fsycl-dump-device-code=">,
+def save_offload_code_EQ : Joined<["-"], "save-offload-code=">,
```

Comment:
Nit: Why not just '-dump-device-code'? Much easier to transition to....

Reply:
`-save-offload-code` is modeled after `-save-temps=`, `-save-stats=` etc.
It also describes the action we are trying to do, which is saving the offload code.
'save' is more aligned with existing LLVM terminology for naming options.